File: Structural/Proxy/Python/real_video_service.py
# real_video_service.py

import time
from video_service import VideoService

class RealVideoService(VideoService):
  def load_video(self, video_name: str):
    print(f"Loading video: {video_name}")
    time.sleep(2)  # Simulating video loading delay
    print(f"Playing video: {video_name}")

File: Structural/Proxy/Java/RealVideoService.java
// RealVideoService.java

public class RealVideoService implements VideoService {
  @Override
  public void loadVideo(String videoName) {
    System.out.println("Loading video: " + videoName);
    try {
      Thread.sleep(2000); // Simulating video loading delay
    } catch (InterruptedException e) {
      e.printStackTrace();
    }
    System.out.println("Playing video: " + videoName);
  }
}
